Integrations/integration-McAfeeNSM.yml

```diff
@@ -0,0 +1,679 @@
+            contextPath : 'NSM.Alerts(val.ID && val.ID == obj.ID)',
+            title : title,
+            data : [
+                {to : 'ID', from : 'uniqueAlertId'},
+                {to : 'Name', from : 'name'},
+                {to : 'State', from : 'alertState'},
+                {to : 'CreatedTime', from : 'summary.event.time'},
+                {to : 'Assignee', from : 'assignTo'},
+                {to : 'Description', from : 'description.definition'},
+                {to : 'EventResult', from : 'summary.event.result'},
+
+                {to : 'Event', from : 'summary.event', humanReadable : false},
+                {to : 'Attack.nspId', from : 'description.referance.nspId', humanReadable : false},
+                {to : 'Attack.attackCategory', from : 'description.attackCategory', humanReadable : false},
+                {to : 'Attacker', from : 'summary.attacker', humanReadable : false},
+                {to : 'Target', from : 'summary.target', humanReadable : false},
+                {to : 'MalwareFile', from : 'details.malwareFile', humanReadable : false},
+            ],
+        },
+        undefined, pascalToSpace);
+    }
+
+    function domains_to_entry(title, domains) {
+        return createEntry(domains, {
+            contextPath : 'NSM.Domains(val.ID && val.ID == obj.ID)',
+            title : title,
+            data : [
+                {to : 'ID', from : 'DomainDescriptor.id'},
+                {to : 'Name', from : 'DomainDescriptor.name'},
+            ],
+        },
+        undefined, pascalToSpace);
+    }
+
+    function sensors_to_entry(title, sensors) {
+        return createEntry(sensors, {
+            contextPath : 'NSM.Sensors(val.ID && val.ID == obj.ID)',
+            title : title,
+            data : [
+                {to : 'ID', from : 'sensorId'},
+                {to : 'Name', from : 'name'},
+                {to : 'Description', from : 'Description'},
+                {to : 'DomainID', from : 'DomainID'},
+                {to : 'IPSPolicyID', from : 'IPSPolicyID'},
+                {to : 'IP Address', from : 'sensorIPAddress'},
+            ],
+        },
+        undefined, pascalToSpace);
+    }
+
+    function login() {
+        var cmd_url = "session";
+        var credentials = Base64.encode(params.credentials.identifier + ':' + params.credentials.password);
+        var headers = {'NSM-SDK-API' : [credentials]};
+        var res = request("GET", cmd_url, headers, '');
+
+        // store current credentials
+        session_credentials = Base64.encode(res.session + ':' + res.userId);
+
+        return res;
+    }
+
+    function get_attacks(attack_id) {
+        var cmd_url, res;
+        if (attack_id) {
+            if (attack_id.match('^0x[0-9A-Fa-f]{8}$') === null) {
+                return 'Error! Attack ID must be formated as 32-bit hexadecimal number. for example: 0x1234BEEF';
+            }
+
+            cmd_url = 'attack/{0}'.format(attack_id);
```

Comment:
I would use native JS (i.e. `'attack/' +  attack_id`). "ES5" way to do so.

Using format is also ok of course :)

Reply:
Its always amazing that you can see which programming language is the no. 1 choice by a programmer code :)
but this is JS and not python :)